### PR TITLE
Fdl async await crash fix

### PR DESCRIPTION
**Pull request by eldhosembabu**
Crash Fix: Async/await violation for Dynamic Links

Related issue : https://github.com/firebase/firebase-ios-sdk/issues/9612

### CODE REVIEW

File: FirebaseDynamicLinks/Sources/FIRDynamicLinks.m [hunk 1]
```diff
@@ -440,7 +440,11 @@ - (nullable FIRDynamicLink *)dynamicLinkFromCustomSchemeURL:(NSURL *)url {
       }
     }
   }
-  mainQueueCompletion(nil, nil);
+  mainQueueCompletion(
+      nil, [NSError errorWithDomain:@""
```
Comment (ryanwilson) [inline FirebaseDynamicLinks/Sources/FIRDynamicLinks.m:444]: Domain shouldn't be empty - looks like we use `com.firebase.dynamicLinks` for generic errors in `FIRDynamicLinkNetworking.m`, we should use that again here. Or, if you think this is related to durable deep links, you could use `kFirebaseDurableDeepLinkErrorDomain` as its used elsewhere in this file.

File: FirebaseDynamicLinks/Sources/FIRDynamicLinks.m [hunk 2]
```diff
@@ -440,7 +440,11 @@ - (nullable FIRDynamicLink *)dynamicLinkFromCustomSchemeURL:(NSURL *)url {
...
+                               code:0
+                           userInfo:[NSDictionary dictionaryWithObject:@"Failed to parse!"
+                                                                forKey:@"Reason"]]);
```
Comment (ryanwilson) [inline FirebaseDynamicLinks/Sources/FIRDynamicLinks.m:447]: This should be [NSLocalizedDescriptionKey](https://developer.apple.com/documentation/foundation/nslocalizeddescriptionkey?language=objc)

File: FirebaseDynamicLinks/Sources/FIRDynamicLinks.m [hunk 3]
```diff
@@ -440,7 +440,11 @@ - (nullable FIRDynamicLink *)dynamicLinkFromCustomSchemeURL:(NSURL *)url {
       }
     }
   }
-  mainQueueCompletion(nil, nil);
+  mainQueueCompletion(
+      nil, [NSError errorWithDomain:@""
+                               code:0
+                           userInfo:[NSDictionary dictionaryWithObject:@"Failed to parse!"
```
Comment (ryanwilson) [inline FirebaseDynamicLinks/Sources/FIRDynamicLinks.m:446]: "Universal link URL could not be parsed by Dynamic Links."

File: FirebaseDynamicLinks/Sources/FIRDynamicLinks.m (hunk 1 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/Sources/FIRDynamicLinks.m]: Done..Using "com.firebase.dynamicLinks" now.

File: FirebaseDynamicLinks/Sources/FIRDynamicLinks.m [hunk 4]
```diff
@@ -440,7 +440,11 @@ - (nullable FIRDynamicLink *)dynamicLinkFromCustomSchemeURL:(NSURL *)url {
       }
     }
   }
-  mainQueueCompletion(nil, nil);
+  mainQueueCompletion(
+      nil, [NSError errorWithDomain:@""
+                               code:0
```
[the remark this answers is not in the document]
Comment (eldhosembabu) [reply FirebaseDynamicLinks/Sources/FIRDynamicLinks.m]: Changed to 1

File: FirebaseDynamicLinks/Sources/FIRDynamicLinks.m (hunk 3 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/Sources/FIRDynamicLinks.m]: Done

File: FirebaseDynamicLinks/Sources/FIRDynamicLinks.m (hunk 2 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/Sources/FIRDynamicLinks.m]: Changed to "NSLocalizedFailureReasonErrorKey", let me know if that is good?

**Review by ryanwilson — APPROVED** [review]

File: FirebaseDynamicLinks/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,6 @@
+# v9.0.0
+- [fixed] Async/await violation for Dynamic Links (#9612)
```
[the remark this answers is not in the document]
Comment (eldhosembabu) [reply FirebaseDynamicLinks/CHANGELOG.md]: Done

**Review by ryanwilson — APPROVED** [review]

**Pull request merged**